frontend/public/components/marketplace/kubernetes-marketplace-items.jsx

```diff
@@ -218,66 +315,71 @@ export class MarketplaceTileViewPage extends React.Component {
             iconClass={normalizedIconClass}
             vendor={vendor}
             description={description}
-            onClick={() => openOverlay(item)}
+            onClick={() => this.openOverlay(item)}
           />;
         })}
       </CatalogTileView.Category>
     );
   }
 
+  renderCategoryTiles(category) {
+    const { categories, selectedCategoryId } = this.state;
+    const { subcategories } = category;
+
+    if (selectedCategoryId === 'all') {
+      return _.map(categories, topCategory => {
+        if (topCategory.id === 'all') {
+          return null;
+        }
+        return this.renderCategory(topCategory, false);
+      });
+    }
+
+    if (!_.size(subcategories)) {
+      return this.renderCategory(category, true);
+    }
+
+    return _.map(subcategories, subcategory => this.renderCategory(subcategory, false));
+  }
+
+  renderEmptyState() {
+    return (
+      <EmptyState className="co-catalog-page__no-filter-results">
+        <EmptyState.Title className="co-catalog-page__no-filter-results-title" aria-level="2">
+          No Results Match the Filter Criteria
+        </EmptyState.Title>
+        <EmptyState.Info className="text-secondary">
+          No marketplace items are being shown due to the filters being applied.
+        </EmptyState.Info>
+        <EmptyState.Help>
+          <button type="button" className="btn btn-link" onClick={() => this.clearFilters()}>Clear All Filters</button>
```

Comment:
Bind `clearFilters` to `this` and use

```suggestion
          <button type="button" className="btn btn-link" onClick={this.clearFilters}>Clear All Filters</button>
```

Arrow functions in render tend can be bad for performance, so it's good to avoid them when not needed.